### DISCUSSION TITLE
Serviços públicos baseados em GraphQL x Restful API

### DISCUSSION THREAD

**Discussion by gmoura** [frontendbr/forum#1615] [Discussão]
Olá pessoal, tenho uma duvida/questionamento, talvez até uma "dúvida existencial", á pergunta é:

Se o mercado de desenvolvimento brasileiro está confortável em consumir serviços públicos baseados em [GraphQL](https://graphql.org/)? Levando em conta que estamos acostumados a consumir serviços baseados em Restful API.

Entendo toda a questão de performance do GraphQL e as diferenças entre os dois paradigmas, porém meu questionamento está voltado para: 

> O quão fácil e doloroso é para um desenvolvedor **consumir/utilizar**  serviços públicos, baseados em GraphQL?

Valeu galera

**Comment by tubedude** [comment]
Como um desenvolvedor mediocre eu fico tranquilo em poder copiar e colar uma chamada GraphQL e fazer a API funcionar. Para mim o que mais pega é tratar o resultado. Essas resposta com edges e nodes acho meio complicado.

**Comment by jgcmarins** [comment]
Aqui tem uma lista de APIs GraphQL públicas: https://github.com/APIs-guru/graphql-apis
Acho que quanto mais canais vc tiver, melhor.
A adoção do GraphQL é cada vez maior então, na minha percepção, empresas que já possuem um API REST pública, também deveriam liberar uma API GraphQL pública.
O contrário também é válido.

[Empresas que usam GraphQL no Brasil](https://github.com/graphql-brasil/empresas-que-usam-graphql-no-brasil)

**Comment by NicholasKuchiniski** [comment]
Não acho legal. Obriga quem vai usar a API a adicionar uma "depêndencia" (seja lib, seja conhecimento técnico de alguém) desnecessária no projeto. Desnecessária no sentido de que, não uso GraphQL para nada no meu projeto, e vou ter que usar ele só para consumir uma API.

Tenho visto quem que quem quer consumir APIs com GraphQL, cria um servidor e usa como middleware para fazer as requisições necessárias para uma API baseada em REST (nada impede que você faça isso também, como o @jgcmarins  falou, liberar duas APIs).
